### PR TITLE
Add ThreadPool.QueueUserWorkItem(..., bool preferLocal)

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/corefx/issues/12442
cc: @kouvel, @tarekgh, @benaadams, @geoffkizer

### CODE REVIEW

**Comment by benaadams** [comment]
Does it need exposing in reference apis somewhere?

**Comment by stephentoub** [comment]
> Does it need exposing in reference apis somewhere?

In corefx, once this is merged and the relevant build is consumed into corefx (and once the same thing happens for the corresponding change in corert).

**Comment by benaadams** [comment]
LGTM

File: src/mscorlib/src/System/Threading/ThreadPool.cs [hunk 1]
```diff
@@ -1268,9 +1268,12 @@ bool executeOnlyOnce
         }
 
         public static bool QueueUserWorkItem(WaitCallback callBack) =>
-            QueueUserWorkItem(callBack, null);
+            QueueUserWorkItem(callBack, null, preferLocal: false);
 
-        public static bool QueueUserWorkItem(WaitCallback callBack, object state)
+        public static bool QueueUserWorkItem(WaitCallback callBack, object state) =>
+            QueueUserWorkItem(callback, state, preferLocal: false);
```
Comment (benaadams) [inline src/mscorlib/src/System/Threading/ThreadPool.cs]: Should be `callBack` caps on `B`

**Pull request merged**